tests/unit-tests/test_utils.py

```diff
@@ -1237,6 +1239,41 @@ def test_get_graph_name():
         assert graph_name == 'dummy'
 
 
+def create_dummy_file(file_path):
+    """ Create an empty dummy file for testing
+    """
+    if os.path.exists(file_path):
+        return
+
+    with open(file_path, 'w') as f:
+        f.close()
+
+def test_wrap_model_artifacts():
+    """
+    """
+    # with tempfile.TemporaryDirectory() as tmpdirname:
+    tmpdirname = '/tmp/wrap_model_artifacts/'
+    os.makedirs(tmpdirname, exist_ok=True)
+    entry_path = os.path.join(tmpdirname, 'nc_infer_entry.py')
+    model_path = os.path.join(tmpdirname, 'model.bin')
+    yaml_path = os.path.join(tmpdirname, 'test.yaml')
+    json_path = os.path.join(tmpdirname, 'test.json')
+
+    create_dummy_file(entry_path)
+    create_dummy_file(model_path)
+    create_dummy_file(yaml_path)
+    create_dummy_file(json_path)
+
+    output_path = '/tmp/output_folder/'
+    os.makedirs(output_path, exist_ok=True)
+    output_file = wrap_model_artifacts(model_path, yaml_path, json_path, entry_path,
+                                       wrap_name='model', output_path=output_path)
+    assert os.path.exists(os.path.join(output_file))
```

Comment:
You should also test all the failure cases.

Reply:
added all failure cases.